### DISCUSSION TITLE
MIDI usb upload speed

### DISCUSSION THREAD

**Discussion by conversy** [christofmuc/KnobKraft-orm#196] [Ideas]
Hi,

I experienced some issues when uploading sysex files on my Solaris unit.
One hypothesis is the transmission is too fast for the solaris usb stack.
Is it possible to throttle the usb-midi connection speed?

Best,

Stéphane

**Comment by christofmuc** [comment]
There is a Python callback you can implement to return a message delay number, which is somewhat respected by the code.

05.01.2023 21:09:48 conversy ***@***.***>:

**Reply by conversy** [reply]
is it `generalMessageDelay`?
Sorry, I overlooked it...

**Reply by christofmuc** [reply]
Yes, though I am not sure if it works in all places. Give it a try. Worst case you could also make the Python program sleep before providing a message. Timing sometimes is tricky

**Comment by christofmuc** [comment]
Closing as answered.
